src/fsharp/TypeChecker.fs

```diff
@@ -5316,12 +5316,14 @@ and TcPat warnOnUpper cenv env topValInfo vFlags (tpenv, names, takenNames) ty p
                     for (id, pat) in pairs do
                         match argNames |> List.tryFindIndex (fun id2 -> id.idText = id2.idText)  with
                         | None -> 
-                            let caseName = 
-                                match item with
-                                | Item.UnionCase(uci, _) -> uci.Name
-                                | Item.ExnCase tcref -> tcref.DisplayName
-                                | _ -> failwith "impossible"
-                            error(Error(FSComp.SR.tcUnionCaseConstructorDoesNotHaveFieldWithGivenName(caseName, id.idText), id.idRange))
+                            match item with
+                            | Item.UnionCase(uci, _) ->
+                                error(Error(FSComp.SR.tcUnionCaseConstructorDoesNotHaveFieldWithGivenName(uci.Name, id.idText), id.idRange))
+                            | Item.ExnCase tcref -> 
+                                error(Error(FSComp.SR.tcExceptionConstructorDoesNotHaveFieldWithGivenName(tcref.DisplayName, id.idText), id.idRange))
+                            | _ ->
+                                error(Error(FSComp.SR.tcConstructorDoesNotHaveFieldWithGivenName(id.idText), id.idRange))
```

Comment:
But this case is impossible, correct? Shouldn't we report this somehow?

Reply:
"impossible"

Reply:
What I mean is that `item` is matched some lines above to be one of the two cases and `item` is immutable? This case means someone has changed something and missed to do some work. Removing `failwith` means the mistake will be spotted later (or not at all)

Reply:
actually that is what already happend. the case was added and nobody changed the failwith "impossible"

Reply:
But line above reads `| (Item.UnionCase _ | Item.ExnCase _) as item ->` how can this match another case?

Reply:
it does not today. wait until someone adds the active pattern part ;-)